First intermediary version - basic function, connectivity and objects for runs

matthias-dirickx opened the pull request:
First intermediary version - basic function, connectivity and objects for runs

Not included: infrastructure to get the test instance ID from the test, test config or other place.

for runs to be meaningful we still need:
- links to other qc objects - question how to manage
- Support for attachments

Commit:
updated tests - coverage for defect now. client itself not functionally covered yet.

matthias-dirickx committed:
Delete vcs.xml

matthias-dirickx committed:
Delete modules.xml

matthias-dirickx committed:
Delete misc.xml

matthias-dirickx committed:
Delete encodings.xml

matthias-dirickx committed:
Delete compiler.xml

matthias-dirickx committed:
Delete .name

matthias-dirickx committed:
Delete scope_settings.xml

matthias-dirickx committed:
Delete Maven__javax_annotation_javax_annotation_api_1_2.xml

matthias-dirickx committed:
Delete Maven__javax_ws_rs_javax_ws_rs_api_2_0.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_external_asm_all_repackaged_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_external_cglib_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_external_javax_inject_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_hk2_api_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_hk2_locator_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_hk2_utils_2_2_0_b10.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_hk2_osgi_resource_locator_1_0_1.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_jersey_core_jersey_client_2_1.xml

matthias-dirickx committed:
Delete Maven__org_glassfish_jersey_core_jersey_common_2_1.xml

matthias-dirickx committed:
Delete profiles_settings.xml

Commit:
updated model with projects and domains (retrieve only). Added extra tests for projects and domains + finished test for defect. Created login test class, but not filled yet.

Commit:
Merge branch 'develop' of https://github.com/matthias-dirickx/QualityCenter-Java-Client into develop

Commit:
finished log in log out test - weak test only. General remark: all types of checks and balances are missing still. Need some login logout error handling.

Commit:
Changed Domain class and added test for Domains.

Commit:
Added entities collection, not tested.

matthias-dirickx committed:
Create README.md

matthias-dirickx committed:
Update README.md

Commit:
updated license. Headers now included.

Commit:
Merge branch 'develop' of https://github.com/matthias-dirickx/QualityCenter-Java-Client into develop

Commit:
Added foundation to start working with RUN and RUNSTEP objects. Added entities and fields. Tests not yet added. Also made post generic for entity by including project and domain in the entity object --> Future: remove specific post and put per entity and have one generic post and put (keeps the code clean).

matthias-dirickx committed:
Create LICENSE

matthias-dirickx committed:
Update README.md

Commit:
made put and post generic - expanded and abused enum to get data in there. Added tests, all passing. Need to increase coverage to support future developments and strategize the fluent api.

Commit:
Merge branch 'develop' of https://github.com/matthias-dirickx/QualityCenter-Java-Client into develop

Commit:
added missing header file

matthias-dirickx committed:
Update README.md

matthias-dirickx committed:
Update README.md

Commit:
Added the basis for the entity, entities and implementation in the client plus tests to cover the load.

Commit:
Merge branch 'develop' of https://github.com/matthias-dirickx/QualityCenter-Java-Client into develop

Commit:
Updated readme, Added composits module and interface, removed set method for totalresults as this is never used (put or post with multiple items does not need the totalresults - it is only for get, so will never use this in the current intended use.

Commit:
Updated tests and adapted RestCallHandler to have a better method for isLoggedIn(). Now actually checking if logged in instead of returning something based on a static class variable.

Commit:
Updated tests - coverage now ok for all but the exception and the composit type

Commit:
updated pom java version to be 1.12, updated commit statement for commitable to be without arguments. The composits now operate with a static client that comes from a singleton class that is fed with the credentials and host from the properties file.

Commit:
updated the runandrunsteps object to contain methods to access the run fields and setting the run steps object and the run object directly.

tlusk commented:
@matthias-dirickx I'm amazed someone found some use from this little library wrapper I wrote at my old job so long ago.  I don't have anything at all to do with HPQC anymore so I can't verify/validate these changes so I'd recommend just forking and keeping it in your repo (which maybe you had the same thought which is why you closed this pull request).

One thing though, this code at least from the point that you forked it did not have a LICENSE at all (my bad) but that means technically without a license the code is not legal to use in a fork.  I've fixed this and re-licensed this under the MIT license, so you should probably rebase your fork or merge in the changes where I added the license for legal reasons.

Anyway good luck to you.

matthias-dirickx commented:
Thanks alot for adding the license. I did not think of that. I'll try and find out if I can Keep my GNU or I need to changebit to MIT then. Purpose was to keep it free to use and adapt and to encourage implementations to be open source as well.

Good thing that I accidentally tried to merge it in your master I guess. 😊


Thanks for the wishes, and thank you for throwing that on there. It was a good starting point.

Best to you!

Kind regards,

Matthias


Outlook voor Android downloaden<https://aka.ms/ghei36>
________________________________
From: Timothy Lusk <notifications@github.com>
Sent: Tuesday, October 29, 2019 4:38:10 PM
To: tlusk/QualityCenter-Java-Client <QualityCenter-Java-Client@noreply.github.com>
Cc: matthias-dirickx <matthias.dirickx@outlook.com>; Mention <mention@noreply.github.com>
Subject: Re: [tlusk/QualityCenter-Java-Client] First intermediary version - basic function, connectivity and objects for runs (#1)


@matthias-dirickx<https://github.com/matthias-dirickx> I'm amazed someone found some use from this little library wrapper I wrote at my old job so long ago. I don't have anything at all to do with HPQC anymore so I can't verify/validate these changes so I'd recommend just forking and keeping it in your repo (which maybe you had the same thought which is why you closed this pull request).

One thing though, this code at least from the point that you forked it did not have a LICENSE at all (my bad) but that means technically without a license the code is not legal to use in a fork. I've fixed this and re-licensed this under the MIT license, so you should probably rebase your branch or merge in the changes where I added the license for legal reasons.

Anyway good luck to you.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/tlusk/QualityCenter-Java-Client/pull/1?email_source=notifications&email_token=AG7IF36YFINY6YNEYAUAATTQRBKGFA5CNFSM4JFTOFN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECQ7HJQ#issuecomment-547484582>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AG7IF32DMZFPVXA5OA5Z2ALQRBKGFANCNFSM4JFTOFNQ>.

tlusk commented:
@matthias-dirickx because MIT is more permissive you can re-license MIT into GNU as long as you also retain the original MIT license file text with my name somewhere in your repo.  You just can't go the other way (GNU -> MIT).

Here's an example: https://softwareengineering.stackexchange.com/a/368621

I've actually spent the past couple months going through license compliance for the OSS libraries we're currently using at my current job so I've been dealing with a lot of this lately 😛.

Enjoy,
Tim
